src/compiler/checker.ts

```diff
@@ -2028,6 +2028,10 @@ namespace ts {
                         let suggestion: Symbol | undefined;
                         if (suggestedNameNotFoundMessage && suggestionCount < maximumSuggestionCount) {
                             suggestion = getSuggestedSymbolForNonexistentSymbol(originalLocation, name, meaning);
+                            // Avoid suggesting "global" if it refers to a global scope augmentation declaration.
+                            if (suggestion && suggestion.valueDeclaration && isAmbientModule(suggestion.valueDeclaration) && isGlobalScopeAugmentation(suggestion.valueDeclaration)) {

```

Comment:
> you don't need `isAmbientModule` since it's implied by `isGlobalScopeAugmentation`

[`isGlobalScopeAugmentation` takes a `ModuleDeclaration`](https://github.com/microsoft/TypeScript/blob/9d286a21aef56a2d9687198daa23ac224d10ad37/src/compiler/utilities.ts#L748), so `isAmbientModule` is there to narrow `suggestion.valueDeclaration`.

> is it possible to get other ambient modules as suggestions? They all have string names, so I suspect not.

I don't think so. I tried and failed to make it happen, anyway.

Reply:
oh, makes sense. I didn't load this code in an editor.